### ISSUE TITLE
[$40] Allow editing link for final deliverable

### ISSUE THREAD

**Issue by RishiRajSahu** [appirio-tech/connect-app#2492]
Links are not editable once the delivery is completed, but they should be. What would happen if a copilot moves things after they've completed adding links, and there's an error? Keep the option to edit/remove links for manager/copilots/admins at all times.

continued from https://github.com/appirio-tech/connect-app/issues/2301

Aha! Link: https://topcoder.aha.io/features/TCCONNECT-461

**Comment by RishiRajSahu** [comment]
@vic-topcoder can we have some UI cue for final deliverable link to give user an intuition of editing capability ?

**Comment by vic-topcoder** [comment]
@acshields - this is more of a product question - should we allow managers to modify links after delivery to customer?

I think from UX perspective you should keep the editing capability to allow for error recovery. This means keep the same editing mode - pen icon to the right of the link.

**Comment by acshields** [comment]
I think its fine to have the ability to modify links after delivery to the customer. Is a notification triggered to the customer regarding changed links, even after the project is complete? @vikasrohit @vic-topcoder

**Comment by vic-topcoder** [comment]
I'm not sure if we cover any timeline events with notifications. @vikasrohit ?

**Comment by vikasrohit** [comment]
@acshields @vic-topcoder we are incorporating timeline and milestone events in upcoming release, however, those will be limited to milestone completion, milestone activation and milestone waiting for customer input events. It requires more analysis and discussions to raise events on data modification of already completed milestones because it might trigger too many notifications to the customers and we have to very specific when detecting changing in data of milestone which would require highly coupled code which in turn is always difficult to maintain.
However, we would still analyze it in future releases.

fyi @RishiRajSahu

**Comment by vic-topcoder** [comment]
I would think that if a final deliverable link is updated there's a reason for it. Not sure if we're actually changing the package (new version of deliverables - have to notify customer) or just changing the location (same package - customer doesn't need a notification). For the time being, let's not add the notification on editing the final deliverable link. Just focus on making the link editable.

**Comment by acshields** [comment]
I am good with your proposal above @vic-topcoder

**Comment by vikasrohit** [comment]
@RishiRajSahu unblocked the issue as requirements are clear now.

**Comment by PrakashDurlabhji** [comment]
@maxceem please assign to me thanks

**Comment by maxceem** [comment]
As the issue hasn't been completed during 12 hours and no comment was provided, this issue is now open for pick up.

**Comment by gets0ul** [comment]
Can I work on this?

**Comment by PrakashDurlabhji** [comment]
@maxceem i am actively working on this.pls give me 1hr. i will raise PR.

**Comment by PrakashDurlabhji** [comment]
@maxceem i am completed with it shall i raise PR

**Comment by maxceem** [comment]
@PrakashDurlabhji if @gets0ul didn't really start working on this you can finish the issue.

@gets0ul please let us know if you already spent some efforts on this issue or we can leave this issue to finish for @PrakashDurlabhji?

**Comment by maxceem** [comment]
@gets0ul yes, please create PR as you followed the Bug Bash rules.

@PrakashDurlabhji sorry, but as another participant already started working on this issue I cannot give you an exception for this issue. Note, that per rules for this Bug Bash participants has to complete issues during 12 hours or provide the valid reasons for the extension.

**Comment by gets0ul** [comment]
@maxceem PR is done and ready for review.

**Comment by PrakashDurlabhji** [comment]
@maxceem i dont think PR solution is correct. just for raising the PR it has been implemented so that getsoul can take another issue and also this issue is not released from him.

**Comment by maxceem** [comment]
@PrakashDurlabhji I'll check the PR, but so far we don't have any formal rule for the cases when PR doesn't fulfill the requirements properly. If we see that there is a high rate of PR which doesn't accomplish tasks, we could make more strict rules for such cases or give warnings to particular members.

**Comment by gets0ul** [comment]
@PrakashDurlabhji  
Please no need for personal remark. Just follow the rules.  
You were already late for 6 hours without any reason. In fact, the rules allow you to be late, extending your time on an issue as long as reason is provided.  

I waited for about 40 minutes after I submit the PR before I picked another ticket.  

And about the PR, just wait for the copilot/admin to review it and give feedback if needed.

**Comment by maxceem** [comment]
@gets0ul I've checked your PR, please have look https://github.com/appirio-tech/connect-app/pull/2605#pullrequestreview-164484821

**Comment by RishiRajSahu** [comment]
Closing merged to dev.
